Source/ASDisplayNode.mm

```diff
@@ -3079,6 +3079,20 @@ - (void)didEnterPreloadState
 {
   ASDisplayNodeAssertMainThread();
   ASDisplayNodeAssertLockUnownedByCurrentThread(__instanceLock__);
+
+  if (self.automaticallyManagesSubnodes) {
+    // Tell the node to apply its applicable pending layout, if any, so that its subnodes are inserted/deleted
+    // and start preloading right away.
+    //
+    // If this node has an up-to-date layout (and subnodes), calling layoutIfNeeded will be fast.
+    //
+    // If this node doesn't have a calculated or pending layout that fits its current bounds, a measurement pass will occur
+    // (see __layout and _u_measureNodeWithBoundsIfNecessary:).
+    // This scenario should be uncommon, and running a measurement pass here is a fine trade-off because preloading
+    // any time after this point would be late.
+    [self layoutIfNeeded];
```

Comment:
Does this work in the case of a node that doesn't have self.isNodeLoaded yet?

Ideally, we would be able to flush the layout / ASM process without relying on a CoreAnimation call at all. For example, a codepath that could call __layout without relying on -layoutSubviews.

Reply:
I agree. That's why I'm calling the bridged `layoutIfNeeded` implementation of `ASDisplayNode` which is thread-safe and doesn't allocate the backing store if the node is not yet loaded (code [here](https://github.com/TextureGroup/Texture/blob/master/Source/Private/ASDisplayNode%2BUIViewBridge.mm#L448)). I also have an [assertion](https://github.com/TextureGroup/Texture/pull/706/files#diff-4bc6f1a7ed5eaad134d9d60d0f5f1c10R134) in the test specifically for this behavior.